src/tests/functional/test_main.py

```diff
@@ -29,25 +31,234 @@
 
 LOG_FILE = os.path.join(os.path.dirname(os.path.realpath(__file__)), "log.txt")
 
-BIN = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../bin/cbc-binary-analysis")
-
-
+# BIN = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../bin/cbc-binary-analysis")
+#
+# SRC = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../")
 # Clear log file for each run
 open(LOG_FILE, "w").close()
 
 
+@pytest.fixture()
+def auth_token(pytestconfig):
+    """Get API token from command line"""
+    return pytestconfig.getoption("token")
+
+
+def _format_config(auth_token, feed_id):
+    """Configure for most of the test cases in this module."""
+    return (f"""id: cbc_binary_toolkit
+version: 0.0.1
+carbonblackcloud:
+  url: https://defense-eap01.conferdeploy.net
+  api_token: {auth_token}
+  org_key: WNEXFKQ7
+  ssl_verify: True
+  expiration_seconds: 3600
+database:
+  _provider: cbc_binary_toolkit.state.builtin.Persistor
+  location: ":memory:"
```

Comment:
If we have a restart test then we will need to set this to a file. If the same hash is used across multiple testcases then they will start to fail